src/pages/login/login-container.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import { RootState } from '../../store/reducer';
+import { connectContainer } from '../../store/redux-container';
+import { LoginStage, switchLoginStage } from '../../store/login';
+
+import { LoginComponent } from './login-component';
+
+export interface LoginContainerProperties {
+  isLoggingIn: boolean;
+  stage: LoginStage;
+  switchLoginStage: (stage: LoginStage) => void;
+}
+
+export class LoginContainer extends React.Component<LoginContainerProperties> {
```

Comment:
Wasn't sure if I handled this correctly - let me know if there's something wrong in here. Something didn't feel quite right about it, but it may be because I'm just not used to the pattern.